### PR TITLE
Move enum handling into a transformer.

### PR DESCRIPTION
**Pull request by mprobst**

### CODE REVIEW

File: src/enum_transformer.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+ */
+
+import * as ts from 'typescript';
+
+import {createSingleQuoteStringLiteral} from './transformer_util';
+import {hasModifierFlag} from './util';
+
+/** isInNamespace returns true if any of node's ancestors is a namespace (ModuleDeclaration). */
+function isInNamespace(node: ts.Node) {
+  // Must us the original node because node might have already been transformed, with node.parent
```
Comment (evmar) [inline src/enum_transformer.ts]: typo: "use"

File: src/enum_transformer.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ */
+export function enumTransformer(typeChecker: ts.TypeChecker, diagnostics: ts.Diagnostic[]):
+    (context: ts.TransformationContext) => ts.Transformer<ts.SourceFile> {
+  return (context: ts.TransformationContext) => {
+    function visitor<T extends ts.Node>(node: T): T|ts.Node[] {
+      if (!ts.isEnumDeclaration(node)) return ts.visitEachChild(node, visitor, context);
+
+      // TODO(martinprobst): The enum transformer does not work for enums embedded in namespaces,
+      // because TS does not support splitting export and declaration ("export {Foo};") in
+      // namespaces. tsickle's emit for namespaces is unintelligble for Closure in any case, so this
```
Comment (evmar) [inline src/enum_transformer.ts]: typo: unintelligible

File: src/transformer_util.ts [hunk 3]
```diff
@@ -685,3 +685,14 @@ export function isTypeNodeKind(kind: ts.SyntaxKind) {
       kind === ts.SyntaxKind.UndefinedKeyword || kind === ts.SyntaxKind.NullKeyword ||
       kind === ts.SyntaxKind.NeverKeyword || kind === ts.SyntaxKind.ExpressionWithTypeArguments;
 }
+
+/**
+ * Creates a string literal that uses single quotes. Purely cosmetic, but increases fidelity to the
+ * existing test suite.
```
Comment (evmar) [inline src/transformer_util.ts]: Given that you're rebaselining the tests below, is it worth reducing use of this?

Comment (mprobst) [reply src/transformer_util.ts]: It's nice to reduce the diff as we port the code base (even if just during development).  We could reconsider once we're done here.

**Pull request merged**